### PR TITLE
Various build command and tests refactoring

### PR DESCRIPTION
**Pull request by bobwhitelock**
This PR contains various refactoring done prior to implementing the PR I'm about to base off this. There shouldn't be any new functionality here, just various tweaks and refactorings made prior to the next branch; see individual commits for details. This is based off https://github.com/alces-software/metalware/pull/153.

### CODE REVIEW

File: spec/minimal_repo.rb [hunk 1]
```diff
@@ -5,13 +5,7 @@ module MinimalRepo
   class << self
     DIRECTORIES = [
       '.git',
-      'config',
-      'hosts',
-      'genders',
-      'dhcp',
       'files',
```
Comment (WilliamMcCumstie) [inline spec/minimal_repo.rb]: Do we still need to have this array if it only has `.git` and `files`? Can this be done the same way as `config`, `hosts`, etc?

Comment (bobwhitelock) [inline spec/minimal_repo.rb]: We need it for those as we only want to create an empty directory in those cases. I guess we could specify a file we don't actually want in those directories and then have `create_files_at` implicitly create the directory as well, but it seems that would make things slightly confusing in order to remove a couple of lines of code.

Comment (bobwhitelock) [inline spec/minimal_repo.rb]: Actually I may be able to do this simply, let me just try it out and see if it looks nicer.

Comment (bobwhitelock) [inline spec/minimal_repo.rb]: I've made this change, let me know what you think: https://github.com/alces-software/metalware/commit/86b15e8acc04ca284e520d5db5f6b7091b135dec.

**Comment by bobwhitelock** [comment]
Made the change, and I've also rebased this on the latest `develop` to resolve the conflicts with the recently merged branch.

**Comment by WilliamMcCumstie** [comment]
Thanks for that change, much simpler now

**Pull request merged**